lib/conjur/puppet_module/http.rb

```diff
@@ -1,50 +1,50 @@
 # frozen_string_literal: true
 
-require 'conjur/puppet_module/ssl'
+require_relative 'ssl'
 
 module Conjur
   module PuppetModule
     # This module is in charge of interacting with the Conjur endpoints
     module HTTP
       class << self
         def get(host_url, path, ssl_certificate, token)
-          uri, use_ssl = parse_url(host_url, path)
-          certs = Conjur::PuppetModule::SSL.load(ssl_certificate)
-
           headers = {}
           if token
             encoded_token = Base64.urlsafe_encode64(token)
             headers['Authorization'] = "Token token=\"#{encoded_token}\""
           end
 
-          Net::HTTP.start uri.host, uri.port, use_ssl: use_ssl, cert_store: certs do |http|
-            response = http.get(uri.request_uri, headers)
+          start_http(host_url, path, ssl_certificate) do |http, uri|
+            http.get(uri.request_uri, headers)
+          end
+        end
 
-            raise Net::HTTPError.new response.message, response unless response.code.match?(%r{^2})
+        def post(host_url, path, ssl_certificate, data, token = nil)
+          headers = {}
+          if token
+            headers['Authorization'] = "Token token=\"#{token}\""
```

Comment:
Does the token not need encoding for post methods ?

Reply:
Nope - the post token doesn't need base64 encoding. It made no sense to me either tbh but it was [in the API docs](https://docs.conjur.org/Latest/en/Content/Developer/Conjur_API_Create_Host.htm?tocpath=Developer%7CREST%C2%A0APIs%7C_____21#Examplewithcurlandjq) and it only worked unencoded 🙃